### PR TITLE
bigtable/table/createReadStream: allow early ending

### LINKED ISSUE #2271: Bigtable this.end() does not stop the stream?
Using Bigtable 0.9.1 - I seem to be getting more than 1 result after calling `this.end()`. Is this expected behavior?

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #2271

### CODE REVIEW

**Comment by stephenplusplus** [comment]
That doesn't seem right, @coveralls.

**Comment by stephenplusplus** [comment]
This is turning out to be a bit of a problem. Marking as `do not merge` until I can get to the bottom of this.

File: packages/bigtable/system-test/bigtable.js [hunk 1]
```diff
@@ -510,7 +509,7 @@ describe('Bigtable', function() {
       it('should get rows', function(done) {
         TABLE.getRows(function(err, rows) {
           assert.ifError(err);
-          assert.strictEqual(rows.length, 4);
+          assert(rows.length >= ROWS.length);
```
Comment (jgeewax) [inline packages/bigtable/system-test/bigtable.js]: Why >= from strict equal? Could the rows returned be more than the rows provided?

Comment (stephenplusplus) [inline packages/bigtable/system-test/bigtable.js]: This PR will need reworking, but currently some tests insert rows, which affects the state outside of the test block. 4 came from "3 in the set up hook, and 1 from the last test". If this test was run in isolation, that last test wouldn't have inserted a row, so we would have 3 not 4. Testing for "at least as many in the set up hook" is an easier solution than creating a table for each test.

**Comment by stephenplusplus** [comment]
Carried on in https://github.com/googleapis/nodejs-bigtable/pull/18

**Pull request closed without merging**